subprojects/packagefiles/frr/meson-add-dependency-definition.patch

```diff
@@ -0,0 +1,167 @@
+From cd9debc344deb17723b5680e7dacde23d7d9e919 Mon Sep 17 00:00:00 2001
+From: Maxime Leroy <maxime@leroys.fr>
+Date: Fri, 23 May 2025 17:44:02 +0200
+Subject: [PATCH] meson add dependency definition
+
+Signed-off-by: Maxime Leroy <maxime@leroys.fr>
+---
+ frr_install.sh    | 35 +++++++++++++++++++
+ meson.build       | 89 +++++++++++++++++++++++++++++++++++++++++++++++
+ meson_options.txt |  7 ++++
+ 3 files changed, 131 insertions(+)
+ create mode 100644 frr_install.sh
+ create mode 100644 meson.build
+ create mode 100644 meson_options.txt
+
+diff --git a/frr_install.sh b/frr_install.sh
+new file mode 100644
+index 0000000..1e40167
+--- /dev/null
++++ b/frr_install.sh
+@@ -0,0 +1,35 @@
++#!/bin/sh
++set -e
++
++frr_srcdir=${1?frr srcdir}
++frr_installdir=${2?frr installdir}
++
++# Create config dir
++mkdir -p "$frr_installdir/etc/frr"
++mkdir -p "$frr_installdir/var/lib/frr"
++
++mkdir -m 0755 -p "$frr_installdir/var/log/frr"
++mkdir -m 0700 -p "$frr_installdir/var/lib/frr"
++
++# Copy main config files
++cp "$frr_srcdir/tools/etc/frr/frr.conf" "$frr_installdir/etc/frr/"
++cp "$frr_srcdir/tools/etc/frr/daemons" "$frr_installdir/etc/frr/"
++
++# Optional: also copy vtysh.conf (used to control integrated config)
++if [ -f "$frr_srcdir/tools/etc/frr/vtysh.conf" ]; then
++    cp "$frr_srcdir/tools/etc/frr/vtysh.conf" "$frr_installdir/etc/frr/"
++fi
++
++# Optional: watchfrr.conf (used by watchfrr daemon if present)
++if [ -f "$frr_srcdir/tools/etc/frr/watchfrr.conf" ]; then
++    cp "$frr_srcdir/tools/etc/frr/watchfrr.conf" "$frr_installdir/etc/frr/"
++fi
++
++# Add a log file for zebra
++log_path="$frr_installdir/var/log/frr/zebra.log"
++conf_file="$frr_installdir/etc/frr/daemons"
++
++sed -i -e "/^zebra_options=/{
++    /--log[[:space:]]*$(printf '%s' "file:$log_path" | sed 's/[\/&]/\\&/g')/! \
++        s|\"$| --log file:$log_path\"|
++}" "$conf_file"
+diff --git a/meson.build b/meson.build
+new file mode 100644
+index 0000000..80bb2c8
+--- /dev/null
++++ b/meson.build
+@@ -0,0 +1,89 @@
++# SPDX-License-Identifier: BSD-3-Clause
++# Copyright (c) 2025 Maxime Leroy, Free Mobile
++
++project('frr', 'c', version : '10.3', license : 'GPL-2.0-or-later')
++
++frr_srcdir   = meson.source_root() / 'subprojects' / 'frr'
++frr_builddir = meson.build_root()  / 'subprojects' / 'frr'
```

Comment:
Instead of this, can you use the following?

https://mesonbuild.com/Reference-manual_builtin_meson.html#mesoncurrent_source_dir

and 

https://mesonbuild.com/Reference-manual_builtin_meson.html#mesoncurrent_build_dir